### ISSUE TITLE
Correct loss gradient calculated on weight 

### ISSUE THREAD

**Issue by nish-ant** [maziarraissi/backprop#1]
As explained in the reader, the gradient of the loss function calculated on weights must involve the matrix multiplication of the activation function with the cost function gradient.

This means that the lines where `loss_weights` is calculated in, for example, `PySimpleNN.py` must be changed from:
            
    loss_weights[l] = np.dot(H[l].T, G[l])
    .
    .
    loss_weights[0] = np.dot(H[0].T, G[0])

to:

    loss_weights[l] = np.dot(A[l].T, G[l])
    .
    .
    loss_weights[0] = np.dot(A[0].T, G[0])
